WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -417,14 +430,23 @@ extension MainTabBarController {
         }
     }
 
-    static func presentPayments() {
+    @discardableResult
+    static func presentPayments() -> InPersonPaymentsMenuViewController? {
         switchToHubMenuTab()
 
         guard let hubMenuViewController: HubMenuViewController = childViewController() else {
-            return
+            return nil
         }
 
-        hubMenuViewController.showPaymentsMenu()
+        return hubMenuViewController.showPaymentsMenu()
+    }
+
+    static func presentCollectPayment() {
+        let viewController = presentPayments()
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            viewController?.openSimplePaymentsAmountFlow()
```

Comment:
Is the `DispatchQueue` async operation needed here? As we're calling `MainTabBarController.presentCollectPayment()` in an async block marked as `@MainActor`

Reply:
Yes, I added a comment to specify why we have that. I will add an issue to fix it in the future